pkg/tsdb/cloudmonitoring/annotation_query.go

```diff
@@ -19,21 +19,46 @@ func (e *CloudMonitoringExecutor) executeAnnotationQuery(ctx context.Context, ts
 		return nil, err
 	}
 
-	queryRes, resp, _, err := queries[0].run(ctx, tsdbQuery, e)
+	queryRes, resp, executedQuery, err := queries[0].run(ctx, tsdbQuery, e)
 	if err != nil {
 		return nil, err
 	}
 
+	err = queries[0].parseResponse(queryRes, resp, executedQuery)
+	if err != nil {
+		queryRes.Error = err
+	}
+
 	metricQuery := firstQuery.Model.Get("metricQuery")
 	title := metricQuery.Get("title").MustString()
 	text := metricQuery.Get("text").MustString()
 	tags := metricQuery.Get("tags").MustString()
+
+	transformToDataframes(queryRes, title, tags, text)
+
 	err = queries[0].parseToAnnotations(queryRes, resp, title, text, tags)
 	result.Results[firstQuery.RefId] = queryRes
 
 	return result, err
 }
 
+func transformToDataframes(queryRes *tsdb.QueryResult, title string, tags string, text string) {
```

Comment:
Since queries[0].parseResponse(queryRes, resp, executedQuery) was called earliser, this is already in formated as data frames. 

You can take a look at how Will did this in his CloudWatch pr 
https://github.com/grafana/grafana/blob/0e4b7555177dca9301b147edde1c685e973c7a61/pkg/tsdb/cloudwatch/annotation_query.go#L117-L136

Reply:
Removed this line, since it may do the job, but annotations are treated differently currently. I'll raise an issue to investigate how we can consolidate those processes :)